src/coreclr/src/jit/gentree.cpp

```diff
@@ -12898,6 +12905,92 @@ GenTree* Compiler::gtFoldExprSpecial(GenTree* tree)
     return op;
 }
 
+//------------------------------------------------------------------------
+// gtFoldBoxNullable -- optimize a boxed nullable feeding a compare
+//
+// Arguments:
+//   tree - binop tree to potentially optimize
+//
+// Return value:
+//   Tree (possibly modified below the root).
+//
+GenTree* Compiler::gtFoldBoxNullable(GenTree* tree)
+{
+    assert(tree->OperKind() & GTK_BINOP);
+    genTreeOps const oper = tree->OperGet();
+
```

Comment:
This method should either assert or check that oper is GT_EQ, GT_NE or GT_GT rather than implicitly relying on the caller to do this check.